server/queries/feedQuery.js

```diff
@@ -10,15 +10,16 @@ LEFT JOIN users u2 ON u2.id = c.author_id
 
 UNION ALL
 
-SELECT g.image_url as post, g.title as title,g.created_at as post_date, g.id as postId, c.comment_id as comment_id, c.comment as comment, c.created_at as date, CONCAT(u.first_name, ' ', u.last_name) as post_author,u2.profile_pix as comment_author_profile, u2.first_name as comment_author,u2.last_name as comment_author_last_name,u.jobrole as author_jobrole, u.profile_pix,
+SELECT g.image_url as post, g.title as title,g.created_at as post_date, g.id as postId, c.id as comment_id, c.comment as comment, c.created_at as date, CONCAT(u.first_name, ' ', u.last_name) as post_author,u2.profile_pix as comment_author_profile, u2.first_name as comment_author,u2.last_name as comment_author_last_name,u.jobrole as author_jobrole, u.profile_pix,
 (SELECT COUNT(gif_id) FROM gif_comment WHERE gif_id = g.id) as number_of_commennt,
 (SELECT COUNT(gif_id) FROM gif_likes WHERE gif_id = g.id) as number_of_likes,
 EXISTS(SELECT * FROM gif_likes l WHERE l.gif_id = g.id and l.author_id = $1) AS isLiked
 FROM gifs g
 LEFT JOIN gif_comment c ON c.gif_id = g.id
 LEFT JOIN users u ON u.id = g.user_id
 LEFT JOIN users u2 ON u2.id = c.author_id
-ORDER BY date DESC NULLS LAST, post_date DESC
+ORDER BY post_date DESC, date DESC NULLS LAST
+LIMIT 3 OFFSET $2
```

Comment:
the limit should be dynamic the way you do offset and the default should be maybe at least 10 and at most 15

Reply:
Ok sir. I have done that now